apps/mobile/src/app/(tabs)/activity.tsx

```diff
@@ -1,21 +1,11 @@
-import { Screen } from '@/components/screen/screen';
-import { ActivityFlashList } from '@/features/activity/activity-flashlist';
-import { useTotalActivity } from '@/queries/activity/account-activity.query';
-import { t } from '@lingui/core/macro';
-
-import { Text } from '@leather.io/ui/native';
+import ActivityScreenWithAccount from '@/components/activity/activity-with-account.screen';
+import ActivityScreenWithoutAccount from '@/components/activity/activity-without-account-screen';
+import { CurrentAccountLoader } from '@/core/current-account-provider';
 
 export default function ActivityScreen() {
-  const activity = useTotalActivity();
-  const pageTitle = t`All Activity`;
-
   return (
-    <Screen>
-      <Screen.Header
-        leftElement={null}
-        centerElement={<Text variant="heading05">{t`All Activity`}</Text>}
-      />
-      <ActivityFlashList data={activity} header={<Screen.Title>{pageTitle}</Screen.Title>} />
-    </Screen>
+    <CurrentAccountLoader fallback={<ActivityScreenWithoutAccount />}>
```

Comment:
It's debatable but I'm not sure of the value of letting people see this empty activity screen:

https://github.com/user-attachments/assets/f27767d6-ffcd-4ea9-a78c-38e46a0455fe

I would suggest to instead:
- disable the activity tab button if it's an empty wallet
- hide the button until there is activity 

I'm fine with this also but as mentioned on Slack, it's adding a complication checking for existence of account before calling hooks and I think it's a cleaner code solution to simply not call them without an account ID by not letting users access this area without the required info